integration_test/src/test_runner/mod.rs

```diff
@@ -207,13 +181,36 @@ fn check_array_length(array: &Vec<Value>, length: usize) -> Result<(), Error> {
     Ok(())
 }
 
+fn run_query_test(
+    query: &str,
+    entry: &TestEntry,
+    conn: &Connection<AutocommitOn>,
+) -> Result<(), Error> {
+    let stmt = Statement::with_parent(conn).unwrap();
+    unsafe {
+        match odbc_sys::SQLExecDirectW(
+            stmt.handle() as *mut _,
+            to_wstr_ptr(query),
+            query.len() as i32,
+        ) {
+            SqlReturn::SUCCESS => {}
+            sql_return => {
+                return Err(Error::OdbcFunctionFailed(
+                    "SQLExecDirectW".to_string(),
+                    format!("{:?}", sql_return),
+                ))
+            }
+        }
+        validate_result_set(entry, stmt)
```

Comment:
```suggestion
            SqlReturn::SUCCESS => validate_result_set(entry, stmt)
            sql_return => {
                return Err(Error::OdbcFunctionFailed(
                    "SQLExecDirectW".to_string(),
                    format!("{:?}", sql_return),
                ))
            }
        }
```
I may be misunderstanding this function, but it seems we only validate the result set if we get `SUCCESS`, so why not have this function live within the match statement instead of after it?

Reply:
That's a good point, thanks for catching that.  I moved the function into the match, and it looks cleaner 👍 .  
I also modified the other functions that had the same pattern to move the functionality into the no-op, I must have written this before I had my coffee ☕